cfme/middleware/server.py

```diff
@@ -0,0 +1,53 @@
+from cfme.fixtures import pytest_selenium as sel
+from cfme.web_ui import CheckboxTable
+from cfme.web_ui.menu import nav, toolbar as tb
+from . import LIST_TABLE_LOCATOR, mon_btn, pwr_btn, MiddlewareBase
+
+list_tbl = CheckboxTable(table_locator=LIST_TABLE_LOCATOR)
+
+nav.add_branch(
+    'middleware_servers', {
+        'middleware_server': lambda ctx: list_tbl.select_row('Server Name', ctx['name']),
+        'middleware_server_detail':
+            lambda ctx: list_tbl.click_row_by_cells({'Server Name': ctx['name']}),
+    }
+)
+
+
+class MiddlewareServer(MiddlewareBase):
+    property_tuples = [('name', 'name')]
+
+    def __init__(self, name, provider=None, **kwargs):
+        if name is None:
+            raise KeyError("'name' should not be 'None'")
+        self.name = name
+        self.provider = provider
+        self.product = kwargs['product'] if 'product' in kwargs else None
+        self.id = kwargs['id'] if 'id' in kwargs else None
+
+    @classmethod
+    def servers(cls, provider):
+        sel.force_navigate('middleware_servers')
+        deployments = []
+        for row in list_tbl.rows():
+            deployments.append(MiddlewareServer(
+                provider=provider, name=row.server_name.text, product=row.product.text))
+        return deployments
+
+    def load_details(self, refresh=False):
+        if not self._on_detail_page():
+            sel.force_navigate('middleware_server_detail', context={'name': self.name})
+        if refresh:
+            tb.refresh()
+
+    def reload_server(self):
+        self.load_details(refresh=True)
+        pwr_btn("Reload Server", invokes_alert=True)
```

Comment:
If a button invokes an alert *in reality*, then the automation code **must** handle the alert. invokes_alert only says the botton code to not execute any other selenium calls after click so the alert can be handled. This will otherwise ruin the browser.